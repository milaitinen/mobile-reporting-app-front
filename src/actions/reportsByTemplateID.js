// easier to refactor later and debug
export const STORE_REPORTS_BY_TEMPLATE_ID = 'STORE_REPORTS_BY_TEMPLATE_ID';

export const storeReportsByTemplateID = (reports) => ({
    type: STORE_REPORTS_BY_TEMPLATE_ID,
    reportsByTempID: reports
});
