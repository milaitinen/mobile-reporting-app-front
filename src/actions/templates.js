// easier to refactor later and debug
export const STORE_TEMPLATES = 'STORE_TEMPLATES';

export const storeTemplates = (templates) => ({
    type: STORE_TEMPLATES,
    templates: templates
});
