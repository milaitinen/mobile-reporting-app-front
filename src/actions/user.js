// easier to refactor later and debug
export const INSERT_EMAIL = 'INSERT_EMAIL';
export const INSERT_PASSWORD = 'INSERT_PASSWORD';
export const INSERT_SERVERURL = 'INSERT_SERVERURL';
export const ADD_TEMPLATES = 'ADD_TEMPLATES';

// return an object
export const insertEmail = (address) => ({
    type: INSERT_EMAIL,
    email: address
});

export const insertPassword = (password) => ({
    type: INSERT_PASSWORD,
    password: password
});

export const insertServerUrl = (url) => ({
    type: INSERT_SERVERURL,
    serverUrl: url
});

export const addTemplatesToData = (templates) => ({
    type: ADD_TEMPLATES,
    templates: templates
});