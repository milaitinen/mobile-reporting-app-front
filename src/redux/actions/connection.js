export const TOGGLE_CONNECTION = 'TOGGLE_CONNECTION';

export const toggleConnection = ({ connectionStatus }) => ({
    type: TOGGLE_CONNECTION,
    isConnected: connectionStatus,
});