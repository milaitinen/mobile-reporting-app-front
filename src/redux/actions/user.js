// easier to refactor later and debug
export const INSERT_EMAIL = 'INSERT_EMAIL';
export const INSERT_PASSWORD = 'INSERT_PASSWORD';
export const INSERT_SERVERURL = 'INSERT_SERVERURL';
export const STORE_TEMPLATES = 'STORE_TEMPLATES';
export const STORE_REPORTS = 'STORE_REPORTS';

// return an object
export const insertEmail = (address) => ({
    type: INSERT_EMAIL,
    email: address
});

export const insertPassword = (password) => ({
    type: INSERT_PASSWORD,
    password: password
});

export const insertServerUrl = (url) => ({
    type: INSERT_SERVERURL,
    serverUrl: url
});