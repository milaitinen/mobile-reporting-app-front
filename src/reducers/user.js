/* eslint-disable no-undef */
import { INSERT_EMAIL } from '../actions/user';

const initialState = {
    email: ' ',
    id: 0,
    authenticated: false,
    templates: {},
};

const reducer = (state = initialState, action) => {
    switch (action.type) {
        case INSERT_EMAIL:
            return {
                ...state,
                email: action.email || ''
            };
        default:
            return state;
    }
};

console.log('initialState', initialState);
export default reducer;