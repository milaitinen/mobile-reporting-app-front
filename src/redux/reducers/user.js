/* eslint-disable no-undef */
import { INSERT_USERNAME, INSERT_PASSWORD, INSERT_SERVERURL, INSERT_TOKEN } from '../actions/user';

const initialState = {
    username: null,
    password: null,
    serverUrl: null,
    token: null
};

const userReducer = (state = initialState, action) => {
    switch (action.type) {
        case INSERT_USERNAME:
            return {
                ...state,
                username: action.username || null
            };
        case INSERT_PASSWORD:
            return {
                ...state,
                password: action.password || null
            };
        case INSERT_SERVERURL:
            return {
                ...state,
                serverUrl: action.serverUrl || null
            };
        case INSERT_TOKEN:
            return {
                ...state,
                token: action.token
            };
        default:
            return state;
    }
};

console.log('initialState', initialState);
export default userReducer;
