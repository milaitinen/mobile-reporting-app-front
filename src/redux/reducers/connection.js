import { TOGGLE_CONNECTION } from '../actions/connection';

const initialState = {
    isConnected: true,
};

const connectionReducer = (state = initialState, action) => {
    switch (action.type) {
        case TOGGLE_CONNECTION:
            return {
                ...state,
                isConnected: action.isConnected,
            };
        default:
            return state;
    }
};
console.log('initialStateConnection', initialState);
export default connectionReducer;