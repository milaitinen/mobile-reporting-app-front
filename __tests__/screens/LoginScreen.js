import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';

import connectedLoginScreen, { LoginScreen } from '../../src/screens/LoginScreen';
import * as userTypes from '../../src/redux/actions/user';
import * as connectionTypes from '../../src/redux/actions/connection';

configure({ adapter: new Adapter() });

jest.mock('../../src/components/OfflineNotice', () => 'OfflineNotice');
jest.mock('../../src/navigation/AppNavigation');

describe('Login screen', () => {
    it('renders correctly', () => {

        const mockStore = configureStore();
        const initialState = {};
        const store = mockStore(initialState);
        const tree = renderer.create(<connectedLoginScreen store={store} />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    jest.unmock('../../src/api');
    const api = require.requireActual('../../src/api');

    const navigate = jest.fn();
    const dispatch = jest.fn();
    const navigation = { state: { params: { isEditable: true } }, navigate: navigate };
    const loginScreen = shallow(<LoginScreen navigation={navigation} dispatch={dispatch} />);

    it('does not call dispatch in logIn function when token is undefined', async () => {
        api.login = jest.fn(() => new Promise(resolve => resolve()));

        await loginScreen.find('SignInButton').simulate('press');
        expect(dispatch).not.toHaveBeenCalledWith({ type: userTypes.INSERT_TOKEN, token: true });
    });

    it('calls dispatch in logIn function correctly', async () => {
        api.login = jest.fn(() => new Promise(resolve => resolve(true)));

        await loginScreen.find('SignInButton').simulate('press');
        expect(dispatch).toHaveBeenCalledWith({ type: userTypes.INSERT_TOKEN, token: true });
    });

    // this.props.dispatch(toggleConnection({ connectionStatus: isConnected }));
    it('dispatches toggleConnection when calling handleConnectionChange', () => {
        loginScreen.instance().handleConnectionChange(true);
        expect(dispatch).toHaveBeenCalledWith({ type: connectionTypes.TOGGLE_CONNECTION, isConnected: true });
    });
});
