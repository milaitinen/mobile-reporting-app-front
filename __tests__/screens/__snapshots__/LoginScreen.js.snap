// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen renders correctly 1`] = `
<connectedLoginScreen
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
